tests/test_fewsnet.py

```diff
@@ -0,0 +1,89 @@
+"""Tests for the FewsNet module."""
+import pytest
+
+from aatoolbox.datasources.fewsnet.fewsnet import download_fewsnet
+
+
+@pytest.fixture
+def _mock_download_call(mocker, tmp_path):
+    """
+    Call the download_fewsnet function.
+
+    Use mocked url and unzip call and have option to set output
+    of country and region download functions
+    """
+    FakeDownloadUrl = mocker.patch(
+        "aatoolbox.datasources.fewsnet.fewsnet.download_url"
+    )
+    mocker.patch("aatoolbox.datasources.fewsnet.fewsnet.unzip")
+
+    # this enables us to set the output of the _download_fewsnet_country()
+    # and _download_fewsnet_region functions
+    # the country_data and region_data arguments can be passed
+    # as arg when calling _mock_download_call
+    def _method(country_data, region_data):
+        if not country_data:
+            mocker.patch(
+                "aatoolbox.datasources.fewsnet.fewsnet."
+                "_download_fewsnet_country",
+                return_value=None,
+            )
+        if not region_data:
+            mocker.patch(
+                "aatoolbox.datasources.fewsnet.fewsnet."
+                "_download_fewsnet_region",
+                return_value=None,
+            )
+        output_path = download_fewsnet(
```

Comment:
I feel like this step should take place outside of the fixture, because in principle you might want to test passing different parameters.

Reply:
Get what you mean and tried to refactor. In a sense it became better but imo also a bit messier. Would you be willing to have a look and see if you got a magic idea?

Reply:
Are you sure you pushed all your updates? The latest version doesn't work for me, and also the structure hasn't changed

Reply:
Ah no oops sorry, now it is there..